src/python/pants/option/option_types.py

```diff
@@ -352,6 +352,55 @@ class WorkspacePathOption(_OptionBase[str, _StrDefault]):
     option_type: Any = custom_types.workspace_path
 
 
+class IntOrStrOption(_OptionBase[Union[str, int], _StrDefault]):
```

Comment:
I don't love creating this unusual, ad-hoc option type for a one-off usage. Why not signal "auto" via 0 or -1 or something?

Reply:
Can do; would simplify a lot. I'm generally not a fan of magic values in my own projects, but this was quite complex to get working anyways.